Makefile.include

```diff
@@ -521,3 +521,19 @@ endif
 CFLAGS := $(patsubst -D%,,$(CFLAGS))
 CFLAGS := $(patsubst -U%,,$(CFLAGS))
 CFLAGS += -include '$(RIOTBUILD_CONFIG_HEADER_C)'
+
+# Set a custom channel if needed
+ifneq (,$(filter netdev2_ieee802154,$(USEMODULE)))
```

Comment:
> I was not clear, sorry. I meant all applications requiring network capability. The second remark (about applications external to RIOT) is still valid though.

Still not true. All application requiring network capabilities *and* want to set their channel at compile time would need to include some form of this `ifneq` construct. If they do it via the Makefile.net_common or on their own is still their prerogative.